Mathlib/Order/Sublocale.lean

```diff
@@ -0,0 +1,227 @@
+/-
+Copyright (c) 2025 Christian Krause. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chriara Cimino, Christian Krause
+-/
+import Mathlib.Order.Nucleus
+import Mathlib.Order.SupClosed
+
+/-!
+# Sublocale
+
+Locales are the dual concept to frames. Locale theory is a branch of point-free topology, where
+intuitively locales are like topological spaces which may or may not have enough points.
+Sublocales of a locale generalize the concept of subspaces in topology to the point-free setting.
+
+## TODO
+
+Create separate definitions for `sInf_mem` and `HImpClosed` (also useful for `CompleteSublattice`)
+
+## References
+
+* [J. Picada A. Pultr, *Frames and Locales*][picado2012]
+* https://ncatlab.org/nlab/show/sublocale
+* https://ncatlab.org/nlab/show/nucleus
+-/
+
+variable {X : Type*} [Order.Frame X]
+open Set
+
+/--
+A sublocale is a subset S of a locale X, which is closed under all meets and for any
+s ∈ S and x ∈ X, we have x ⇨ s ∈ S [picado2012].
```

Comment:
Is it worth mentioning why we make `Sublocale` take in a `Frame` and not a `Locale` or is that obvious to everyone (other than me, possibly)?

Oh, I guess the real question is maybe why we don't define `Locale`? And is it worth mentioning `Locale` in `Mathlib.Topology.Category.Locale` and saying something about the difference?

Reply:
Mentioned!